File: Oidc.AuthorizationServer/Models/AllowedScope.cs
using lvl.Ontology;
using lvl.Ontology.Authorization;
using lvl.Ontology.Conventions;
using System.ComponentModel.DataAnnotations;

namespace lvl.Oidc.AuthorizationServer.Models
{
    /// <summary>
    ///     Models access to an API resource
    /// </summary>
    [Schema("oidc")]
    [HiddenFromApi]
    public class AllowedScope : Entity, IAggregateScope<ClientEntity>
    {
        /// <summary>
        ///     Tha name of an api resource.
        /// </summary>
        [Required]
        public string Name { get; set; }
    }
}

File: oidc/src/models/allowed-scope.ts
import { ClientEntity } from './client-entity';
import { Entity, IAggregateScope, Required } from '@lvl/front-end';

export class AllowedScope extends Entity implements IAggregateScope<ClientEntity> {
    @Required() public name: string;

    constructor(options?: IAllowedScopeOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
    }
}

interface IAllowedScopeOptions {
    name?: string;
    id?: number;
}
